acme/acme/challenges.py

```diff
@@ -493,6 +494,11 @@ class TLSSNI01(KeyAuthorizationChallenge):
     # boulder#962, ietf-wg-acme#22
     #n = jose.Field("n", encoder=int, decoder=int)
 
+    def __init__(self, *args, **kwargs):
+        warnings.warn("TLS-SNI-01 is deprecated, and will stop working in February 2019.",
```

Comment:
I don't have a strong preference, but I'd personally like us to remove the date here as things like that can change but a released version of Certbot cannot.